File: Server/src/main/java/com/api/server/foodb/CiteThisTextbook.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "cite_this_textbooks")
public class CiteThisTextbook {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 255)
    @Column(name = "isbn")
    private String isbn;

    @NotNull
    @Lob
    @Column(name = "title", nullable = false)
    private String title;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Size(max = 255)
    @Column(name = "ref_id")
    private String refId;

    @Size(max = 255)
    @NotNull
    @Column(name = "authors", nullable = false)
    private String authors;

    @Size(max = 255)
    @Column(name = "edition")
    private String edition;

    @Size(max = 255)
    @NotNull
    @Column(name = "publisher", nullable = false)
    private String publisher;

    @Size(max = 255)
    @NotNull
    @Column(name = "year", nullable = false)
    private String year;

    @Size(max = 255)
    @Column(name = "book_format")
    private String bookFormat;

    @Size(max = 255)
    @Column(name = "ean")
    private String ean;

}

File: DotNetServer/DotNetServer/Entities/CiteThisTextbook.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Entities;

[Table("cite_this_textbooks")]
[Index("Isbn", Name = "index_cite_this_textbooks_on_isbn")]
public partial class CiteThisTextbook
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("isbn")]
    public string? Isbn { get; set; }

    [Column("title", TypeName = "text")]
    public string Title { get; set; } = null!;

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }

    [Column("ref_id")]
    [StringLength(255)]
    public string? RefId { get; set; }

    [Column("authors")]
    [StringLength(255)]
    public string Authors { get; set; } = null!;

    [Column("edition")]
    [StringLength(255)]
    public string? Edition { get; set; }

    [Column("publisher")]
    [StringLength(255)]
    public string Publisher { get; set; } = null!;

    [Column("year")]
    [StringLength(255)]
    public string Year { get; set; } = null!;

    [Column("book_format")]
    [StringLength(255)]
    public string? BookFormat { get; set; }

    [Column("ean")]
    [StringLength(255)]
    public string? Ean { get; set; }

    [InverseProperty("Textbook")]
    public virtual ICollection<CiteThisTextbookReferencing> CiteThisTextbookReferencings { get; set; } = new List<CiteThisTextbookReferencing>();
}
